ui/packages/catalog-ui-search/src/main/webapp/js/controllers/cesium.layerCollection.controller.js

```diff
@@ -35,6 +35,77 @@ var imageryProviderTypes = {
   SI: Cesium.SingleTileImageryProvider,
 }
 
+const LayerOrdering = {
+  /* params: {prev: array of ids, 
+              cur: array of ids, 
+              layer: id of layer to add}
+    return: new ordering
+    assumption: layer is in cur, cur is a superset of prev, layer ordering has not changed
```

Comment:
❓ Could we add efficient validation regarding the superset property of `cur` at the beginning of each of these ordering helper functions? If verifying that is not cheap, maybe checking **once** in the component itself before all this layer manipulation begins? 

I suspect protecting the utility methods themselves will be fine. I wouldn't expect a map layers configuration to surpass 100 layers.

Reply:
I'll do a check in the add function